### DISCUSSION TITLE
Support adding asset metadata on creation (IOS)

### DISCUSSION THREAD

**Discussion by tiagoacastro** [expo/expo#22023] [Proposals and Ideas]
In the IOS photo app, images show some of the metadata on the information section (the location the image was taken and the caption of the image are two examples). It would be nice to be able to set some of this information when using MediaLibrary's savetolibraryasync or createassetasync. The current use case where I encountered such need was when trying to add images with a description in the caption section.

**Comment by AlimovSV** [comment]
I faced with this issue too, when I use saveToLibraryAsync(uri) it does not copy exif metadata (GPS) from the original photo, but if I use createAssetAsync it does, but requires additional MEDIA_LIBRARY permission. related issue: https://stackoverflow.com/questions/77470854/how-can-i-store-data-in-a-custom-tag-in-exif-with-react-native-and-read-it-with

**Comment by IsaacInsoll** [comment]
Thanks AlimovSV, 

My photos were saving to camera roll but had no metadata (even though the source file did!). 

I swapped `saveToLibraryAsync` with `createAssetAsync` and now all exif information (shutter/aperture etc) is saved with the file in the camera roll.
